src/__tests__/GoEngine_sgf.test.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @jest-environment jsdom
+ */
+
+// ^^ jsdom environment is because getLocation() returns window.location.pathname
+// Same about CLIENT.
+//
+// TODO: move this into a setup-jest.ts file
+
+(global as any).CLIENT = true;
+
+import { TestGoban } from "../TestGoban";
+
+const SGF_TEMPLATES = {
+    unicode:
+        "(;GM[1]FF[4]CA[UTF-8]AP[CGoban:3]ST[2] RU[Japanese]SZ[19]KM[0.00]PW[White]PB[Black]_MOVES_)",
+    honinbo:
+        "(;FF[4]GM[1]SZ[19]AP[]US[]EV[74th Honinbo challenger decision match]PC[]PB[Shibano Toramaru]BR[7d]PW[Kono Rin]WR[9d]KM[6.5]RE[W+1.5]DT[2019-04-10]TM[0]_MOVES_)",
+};
+
+const SGF_MOVES = {
+    unicode: ";W[aa]C[příliš žluťoučký kůň úpěl ďábelské ódy];W[pd]",
+    honinbo: ";B[pd];W[dp];B[pq];W[dc];B[ce];W[cg];B[cq];W[cp];B[dq];W[fq]",
+};
+
+function rmNewlines(txt: string): string {
+    return txt.replace(/[\n\r]/g, "");
+}
+/*
+ * check that parse -> serialize roundtrip generates the same sgf
+ * (at least for the moves)
+ */
+test("sgf -> parseSGF() -> toSGF() roundtrip (moves only)", () => {
+    Object.entries(SGF_TEMPLATES).forEach(([key, template], index) => {
```

Comment:
Nit: you may be interested in [`test.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) which is made for this type of test-cases-in-array pattern.  See an example in @GreenAsJade's PR: https://github.com/online-go/goban/blob/8666815d12dbef1d98c2a6dcf0d3d450cce026c0/src/__tests__/GoMath_positionId.test.ts#L145-L147

Reply:
nice! fixed